### ISSUE TITLE
Some problem about training...

### ISSUE THREAD

**Issue by jwhmicrsoft** [mamonraab/fire-detection-in-video-with-pytroch#3]
Hi, it's me again. I got some new problems when training...

`tensor([[1.],[1.],[0.],[0.]], device='cuda:0')
tensor([[0.5241],[0.5382],[0.5257],[0.5247]], device='cuda:0', grad_fn=<SigmoidBackward>)
loss: 0.688816487789154`

As you can see in the above code, this is the consequence of 30 epochs training. The above tensor `([[1.],[1.],[0.],[0.]])` is the labels means if it is fight video, the tensor below it `([[0.5241],[0.5382],[0.5257],[0.5247]])` is the probability which comes from the sigmoid function.
I don't know why the loss can't drop after 30 epochs training. The loss always is around 0.7. I tried some criterion function such as BCELoss, MSELoss and optimizer using Adam or SGD,but things didn't go well... The loss still didn't drop anyway. Would you like to help me check where the problem is plz? Thank you!

**Comment by jwhmicrsoft** [comment]
And here is my source code.


# Loading the data

```
train_frame = pd.read_csv('train_fight_vid_2.csv', delimiter=',')
test_frame = pd.read_csv('test_fight_vid.csv', delimiter=',')

dataset_train = VideoDataset(train_frame)
dataset_test = VideoDataset(test_frame)

train_loader = DataLoader(dataset=dataset_train, batch_size=4, shuffle=True, num_workers=0)
test_loader = DataLoader(dataset=dataset_test, batch_size=8, shuffle=True, num_workers=0)
```
# build model

```
class Net(nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        num_classes = 1
        dr_rate = 0.2
        pretrained = True
        rnn_hidden_size = 30
        rnn_num_layers = 2
        # get a pretrained vgg19 model (taking only the cnn layers and fine tune them)
        baseModel = models.vgg19(pretrained=pretrained).features
        i = 0
        for child in baseModel.children():
            if i < 26:
                for param in child.parameters():
                    param.requires_grad = False
            else:
                for param in child.parameters():
                    param.requires_grad = True
            i += 1

        num_features = 7680
        self.baseModel = baseModel
        self.dropout = nn.Dropout(dr_rate)
        self.rnn = nn.LSTM(num_features, rnn_hidden_size, rnn_num_layers, batch_first=True)
        self.fc2 = nn.Linear(30, 256)
        self.fc3 = nn.Linear(256, num_classes)

    def forward(self, x):
        batch_size, time_steps, C, H, W = x.size()
        # reshape input to be (batch_size * timesteps, input_size)
        x = x.contiguous().view(batch_size * time_steps, C, H, W)
        x = self.baseModel(x)
        x = x.view(x.size(0), -1)
        # make output as (samples, timesteps, output_size)
        x = x.contiguous().view(batch_size, time_steps, x.size(-1))
        x, (hn, cn) = self.rnn(x)
        x = F.relu(self.fc2(x[:, -1, :]))
        x = self.dropout(x)
        x = self.fc3(x)
        return x
```
# Training

```
model = Net()
device = torch.device('cuda:0')
model.to(device)

criterion = torch.nn.BCELoss(size_average=True)  
# criterion = torch.nn.BCEWithLogitsLoss(weight=None, reduction='mean', pos_weight=None)
# criterion = torch.nn.MSELoss(size_average=False)
# optimizer = torch.optim.SGD(model.parameters(), lr=0.001) 
optimizer = torch.optim.Adam(model.parameters())
# optimizer = torch.optim.Adagrad(model.parameters())

acc = 0
train_num = 0

for epoch in range(50):
    for i, data in enumerate(train_loader):

        inputs, labels = data
        inputs = inputs.to(device)
        labels = labels.to(device)

        fight_pred = model(inputs)

        fight_pred = fight_pred.to(torch.float32)
        fight_pred = torch.sigmoid(fight_pred)
        labels = labels.to(torch.float32)

        labels = labels.view(len(labels), -1)

        loss = criterion(fight_pred, labels)

        optimizer.zero_grad()
        loss.backward()

        optimizer.step()
```

**Comment by mamonraab** [comment]
please send me a colab link  with your data so i can help you debug the problem , i will close the issue since it is not related to the main code
